collections/forms/src/validators/integer.js

```diff
@@ -5,8 +5,14 @@ const invalidIntegerMessage = i18n.t(
     'Please provide a round number without decimals'
 )
 
+// Regex accepts 0 but not a traling zero (i.e 04) as this is rejected by backend
+const INTEGER_PATTERN = /^(-?[1-9]\d*|0)$/
+
 const integer = (value) =>
-    isEmpty(value) || (isNumeric(value) && isInteger(toNumber(value)))
+    isEmpty(value) ||
+    (INTEGER_PATTERN.test(value) &&
+        isNumeric(value) &&
+        isInteger(toNumber(value)))
```

Comment:
This is only indirectly related to the fix you are making, but this PR will introduce some additional confusion so it's probably best if we fix it here...
- `isInteger` is a helper and simply does this `(value) => Number.isSafeInteger(value)`
- This helper is not being used anywhere else in our codebase, it's only used in the `integer` validator above.

So technically we now have two functions that define how we "define and integer" using a slightly different definition. I think we should:
- Remove the `isInteger` helper
- Change the `integer` function to something like this:
```js
const integer = (value) =>
    isEmpty(value) ||
    (isNumeric(value) &&
        INTEGER_PATTERN.test(value) &&
        Number.isSafeInteger(toNumber(value)))
        ? undefined
        : invalidIntegerMessage
```

Reply:
updated